src/De/DeJson.jl

```diff
@@ -2,15 +2,359 @@ module DeJson
+    end
+end
+
+function deser(::NullType, ::Type{Union{Missing,T}}, value_ptr::Ptr{YYJSONVal}) where {T}
+    return if yyjson_is_null(value_ptr)
+        deser(T, value_ptr)
+    end
+end
+
+# NTupleType
+
+function deser(::NTupleType, ::Type{T}, value_ptr::Ptr{YYJSONVal}) where {T<:NamedTuple}
+    return if yyjson_is_obj(value_ptr)
+        (; deser(Dict{Symbol,Any}, value_ptr)...)
+    end
+end
+
+# DictType
+
+function deser(::DictType, ::Type{T}, value_ptr::Ptr{YYJSONVal}) where {N,T<:AbstractSet{N}}
+    return if yyjson_is_arr(value_ptr)
+        T(deser(Vector{N}, value_ptr))
+    end
+end
+
+function deser(::DictType, ::Type{T}, value_ptr::Ptr{YYJSONVal}) where {K,V,T<:AbstractDict{K,V}}
+    return if yyjson_is_obj(value_ptr)
+        iter = YYJSONObjIter()
+        iter_ref = Ref(iter)
+        iter_ptr = Base.unsafe_convert(Ptr{YYJSONObjIter}, iter_ref)
+        GC.@preserve iter begin
+            yyjson_obj_iter_init(value_ptr, iter_ptr) || throw(YYJSONError("Failed to initialize object iterator."))
+            dict_elements = T()
+            for _ = 1:yyjson_obj_size(value_ptr)
+                key_ptr = yyjson_obj_iter_next(iter_ptr)
+                value_ptr = yyjson_obj_iter_get_val(key_ptr)
+                dict_elements[deser(K, key_ptr)] = deser(V, value_ptr)
+            end
+        end
+        dict_elements
+    end
+end
+
+# ArrayType
+
+function deser(::ArrayType, ::Type{T}, value_ptr::Ptr{YYJSONVal}) where {T<:AbstractVector}
+    return if yyjson_is_arr(value_ptr)
+        iter = YYJSONArrIter()
+        iter_ref = Ref(iter)
+        iter_ptr = Base.unsafe_convert(Ptr{YYJSONArrIter}, iter_ref)
+        GC.@preserve iter begin
+            yyjson_arr_iter_init(value_ptr, iter_ptr) || throw(YYJSONError("Failed to initialize array iterator."))
+            array_elements = T(undef, yyjson_arr_size(value_ptr))
+            for i in eachindex(array_elements)
+                elem_ptr = yyjson_arr_iter_next(iter_ptr)
+                array_elements[i] = deser(eltype(T), elem_ptr)
+            end
+        end
+        array_elements
+    end
+end
+
+function deser(::ArrayType, ::Type{T}, value_ptr::Ptr{YYJSONVal}) where {T<:Tuple}
+    return if yyjson_is_arr(value_ptr)
+        if T == Tuple
+            T(deser(Vector{Any}, value_ptr))
+        else
+            iter = YYJSONArrIter()
+            iter_ref = Ref(iter)
+            iter_ptr = Base.unsafe_convert(Ptr{YYJSONArrIter}, iter_ref)
+            GC.@preserve iter begin
+                yyjson_arr_iter_init(value_ptr, iter_ptr) || throw(YYJSONError("Failed to initialize array iterator."))
+                tuple_elements = Vector{Any}(undef, fieldcount(T))
+                for (i, type) in zip(eachindex(tuple_elements), fieldtypes(T))
+                    value_ptr = yyjson_arr_iter_next(iter_ptr)
+                    tuple_elements[i] = deser(type, value_ptr)
+                end
+            end
+            T(tuple_elements)
+        end
+    end
+end
```

Comment:
@gryumov 
In this function, the first condition is intended for pure Tuple, which in advance is unknown by the number of elements
The second branch for tuple {t1, t2, ...} or NTuple